cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandController.java

```diff
@@ -206,6 +209,28 @@ protected List<HasMetadata> doReify(
                 break;
         }
 
+        if (isDatabaseHistorySupported(shardMetadata)) {
+            final Map<String, Object> databaseHistoryConfigs = new LinkedHashMap<>();
+            final String password = new String(Base64.getDecoder().decode(serviceAccountSpec.getClientSecret()),
+                StandardCharsets.UTF_8);
+
+            databaseHistoryConfigs.put("database.history.kafka.bootstrap.servers",
+                connector.getSpec().getDeployment().getKafka().getUrl());
+            databaseHistoryConfigs.put("database.history.kafka.topic", connector.getMetadata().getName() + "-database-history");
+            databaseHistoryConfigs.put("database.history.producer.security.protocol", "SASL_SSL");
+            databaseHistoryConfigs.put("database.history.consumer.security.protocol", "SASL_SSL");
+            databaseHistoryConfigs.put("database.history.producer.sasl.mechanism", "PLAIN");
+            databaseHistoryConfigs.put("database.history.consumer.sasl.mechanism", "PLAIN");
+            databaseHistoryConfigs.put("database.history.producer.sasl.jaas.config",
+                "org.apache.kafka.common.security.plain.PlainLoginModule required username=\""
+                    + serviceAccountSpec.getClientId() + "\" password=\"" + password + "\";");
```

Comment:
In general, password should not be part of a CR but should be referenced from a secret (i.e. https://github.com/bf2fc6cc711aee1a0c2a/cos-fleetshard/blob/0475269fa7989c4d6b07ad2ecdea9272653aef96/cos-fleetshard-operator-debezium/src/main/java/org/bf2/cos/fleetshard/operator/debezium/DebeziumOperandSupport.java#L100-L107)

Reply:
I'll take a look.  I tried to use the secrets configuration reader to pass a reference to the client password from another existing secret but that didn't seem to work, unfortunately. I'll verify if we can use the file reader here.

Reply:
Found a workable solution, checking to verify that its read properly by Kafka Connect.